cli/src/pcluster/templates/cluster_stack.py

```diff
@@ -183,12 +184,36 @@ class FsxConstruct(core.Construct):
     """Create the resources related to the FSX."""
 
     # https://cdkworkshop.com/30-python/40-hit-counter/100-api.html
-    def __init__(self, scope: core.Construct, id: str, fsx: SharedFsx, **kwargs):
+    def __init__(self, scope: core.Construct, id: str, shared_fsx: SharedFsx, subnet_id: str, **kwargs):
         super().__init__(scope, id)
-        self.fsx = fsx
+        self.shared_fsx = shared_fsx
         # TODO add all the other required info other than fsx object and generate template
         # TODO Verify if there are building blocks ready to be used
 
+        fsx.CfnFileSystem(
```

Comment:
Here, we should consider security group ids if using existing fsx-id. As far as I know, If using an existing file system, it must be associated to a security group that allows inbound TCP traffic to port 988. Setting the source to 0.0.0.0/0 on a security group rule provides client access from all the IP ranges within your VPC security group for the protocol and port range for that rule. https://docs.aws.amazon.com/parallelcluster/latest/ug/fsx-section.html.
So, should we add a condition to check fsx_id exists or not?

Reply:
I would demand this responsibility to validators; CDK conversion should only take care of translating the model. Anyway, security groups and subnet ids were missing, so thanks for pointing to this code.